src/nvim/ex_docmd.c

```diff
@@ -8973,6 +8977,15 @@ makeopens (
         return FAIL;
       next_arg_idx = wp->w_arg_idx;
     }
+		
+		if (put_line(fd,"for pair in buffer_names") == FAIL
```

Comment:
Why do you use `pair` and `get()`? And outright incorrect and overly verbose `execute join`: neither join is necessary at all nor you do escaping. I would write that as

```VimL
let s:buffer_names = []
…
call add(…)
…
for [s:src, s:tgt] in s:buffer_names
  if bufexists(s:src)
    execute 'buffer' fnameescape(s:src)
    execute 'file' fnameescape(s:tgt)
  endif
endfor
unlet s:buffer_names s:src s:tgt
```

(And I do not understand now where this code goes, but should not it save what is the current buffer number and do `execute 'buffer' s:saved_curbuf` afterwards?)